warehouse/query-core/src/main/java/datawave/query/transformer/GroupingTransform.java

```diff
@@ -69,13 +75,45 @@ private Key createAttrKey(String fieldName) {
         return new Key(fieldName);
     }
     
+    /**
+     * Aggregate items from the incoming iterator and supply them via the flush method
+     * 
+     * @param in
+     *            an iterator source
+     * @return the flushed value that is an aggregation from the source iterator
+     */
+    public Iterator<Entry<Key,Document>> getGroupingIterator(final Iterator<Entry<Key,Document>> in) {
+        
+        return new Iterator<Entry<Key,Document>>() {
+            
+            Entry<Key,Document> next;
+            
+            @Override
+            public boolean hasNext() {
+                while (in.hasNext()) {
+                    GroupingTransform.this.apply(in.next());
+                }
+                next = GroupingTransform.this.flush();
```

Comment:
The while loop aggregates all of the Documents that are fetchedby the tserver. 
It's also possible to modify this code set an arbitrary limit on the loop so that smaller 'batches' of documents are returned. The benefit would be to reduce wait time for the webserver and to prevent potential OOM errors on the tserver.